crates/evm/execution-types/src/execution_outcome.rs

```
@@ -69,27 +73,31 @@ impl ExecutionOutcome {
     ///
     /// This constructor initializes a new `ExecutionOutcome` instance with the provided
     /// bundle state, receipts, first block number, and EIP-7685 requests.
-    pub const fn new(
+    pub fn new(
+        chain_id: Option<u64>,
         bundle: BundleState,
         receipts: Receipts,
         first_block: BlockNumber,
         requests: Vec<Requests>,
     ) -> Self {
-        Self { bundle, receipts, first_block, requests }
+        let chain_id = chain_id.unwrap_or(ETHEREUM_CHAIN_ID);
```

Safer to just fail instead of accepting None I think